### PR TITLE
enhance: use GPU pool for gpu tasks (#29678)

### PR DESCRIPTION
**Pull request by yah01**
- this much improve the performance for GPU index
- this also reduce 1x copy while parsing index meta
pr: #29678

### CODE REVIEW

**Comment by samhuang-z** [comment]
rerun ut

File: pkg/util/paramtable/component_param.go [hunk 1]
```diff
@@ -1936,6 +1937,13 @@ Max read concurrency must greater than or equal to 1, and less than or equal to
 	}
 	p.MaxReadConcurrency.Init(base.mgr)
 
+	p.MaxGpuReadConcurrency = ParamItem{
+		Key:          "queryNode.scheduler.maGpuReadConcurrency",
+		Version:      "2.0.0",
```
Comment (congqixia) [inline pkg/util/paramtable/component_param.go:1942]: Version shall be 2.3.4?

**Review by congqixia — APPROVED** [review]
/lgtm

**Pull request merged**